tasks/run-local.sh

```diff
@@ -176,7 +176,9 @@ EOF
         -e AMQP_SERVER=$AMQP_POD \
         -e S3_LOGS_URL=$S3_URL_POD/logs/ \
         -e SKIP_STATIC_CHECK=1 \
-        quay.io/cockpit/tasks:${TASKS_TAG:-latest} ${INTERACTIVE:+sleep infinity}
+        quay.io/cockpit/tasks:${TASKS_TAG:-latest} bash
+
+    podman exec -i cockpituous-tasks setup-tasks
```

Comment:
Maybe -- that's what I'm not yet clear about. E.g. do we want to run it in the per-job instance containers? Do they need the bots checkout? (I figure that part is the step child). I think we need to talk about that.

In general I'm sympathetic about making this an entry point, I thought about it myself, too.

Reply:
My gut feeling: Make `setup-tasks` an entry point, but revert "Move initial bots clone to setup-tasks" and live with the duplication in run-local.sh. Then this only does secrets (symlink/OpenShift unpacking) setup, which we really want everywhere.

Reply:
My gut feeling: I hate `setup-tasks`.  I think we have three good tools at our disposal and we need to use them:

 - the container image itself
 - the ability to arbitrarily bind-mount stuff in place
 - the fact that we write almost all of the software that ultimately is running inside of our container and interacting with these config bits

We should be able to construct our entire "world" out of the first two, and to the extent that its not possible, deal with it by way of the last one.

Reply:
Add to the above list, of course: 
 - environment variables

Reply:
- We can replace the `npm config`s with a ~/.npmrc which we bind-mount as a volume, and keep the custom registry in cockpitous.git for our bots; that part is relatively simple
- We can directly mount ~/.config/cockpit-dev/s3-keys and  ~/.config/github-token on `podman` based bots.
- `~/.git-credentials` could be replaced with a volume if we duplicate the secret and its setup as a bind mount.

That leaves the OpenShift user and credentials setup, which *really* needs to be dynamic. We can split it out into `setup-openshift` and start the container through it as custom entry point (or command) in the yaml.

As I said, we can move bots back, we presumably don't need it in the job runners, just the monitor.

Reply:
I think `.git-credentials` isn't actually used, for the record.  My local testing showed that bots (ie: `tasks/__init__.py`) is smart enough to setup the credentials for itself based on the API key from the usual file.

I have no idea what the OpenShift stuff is for so I'm happy to defer to you on that point, but ya, my instinct there was that this is the sort of thing that can be handled in a script elsewhere, if it's required.

And ya, for npm, I just played with that and got:

```
fetch-retries=6
fetch-timeout=600000
fetch-retry-mintimeout=60000
maxsockets=3
registry=https://xyz
```

in my `~/.npmrc`, but also notice that `/etc/npmrc` is a thing, and this is probably a better fit.  I bet we can get the certificate specified via this file, too...

Which brings me to my next point: I wonder if we should have something like /run/external-config/ with files like `npmrc` in it, and symlink those from /etc in the container image.  This would be our primary mechanism for injecting external configuration for everything but secrets (in the `job-runner` sense of the term).  We don't need to do this now, but in the long-run it's probably going to be somewhat cleaner than a dozen bindmounts...

Reply:
>  I bet we can get the certificate specified via this file, too...

That unfortunately needs the silly `$NODE_EXTRA_CERTS` env variable, but that's already handled differently. I.e. we can put that cert file wherever we like (including `/run/external-config/`), just need to point to it.

I'm not sure about .git-credentials. It may be a leftover from the time when we did `npm-update`, `po-refresh` etc. bots in our CI, as we do the [exact same thing in the workflows](https://github.com/cockpit-project/cockpit/blob/main/.github/workflows/weblate-sync-po.yml#L45). So we can try to drop it, and put it back if necessary :grin:

Reply:
Tested on a bot:
```
git clone https://github.com/cockpit-project/starter-kit
cd starter-kit
rpm -rf ~/.npm
unset NODE_EXTRA_CERTS
npm install  # → hangs

npm config set cafile /secrets/npm-registry.crt
npm install # → works
```

Reply:
You can also set `ca` directly as the content of the certificate and avoid having to bring it in as a secret..

Reply:
setup-tasks is gone, and I implemented the things discussed above.